cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -149,6 +178,27 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     return new DataprocClient(new DataprocConf(conf, network, subnet), client, compute, useInternalIP);
   }
 
+  private static PeeringState peeringState(String systemNetwork, String systemProjectId, Network networkInfo) {
```

Comment:
the system network is nullable. It is null when CDAP is not running on GCP. When null, there is no peering between CDAP and the configured network.

Reply:
added `@Nullable` annotation and corresponding check.